lib/tapioca/dsl/compilers/active_model_validations_confirmation.rb

```diff
@@ -0,0 +1,97 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "active_model"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Dsl
+    module Compilers
+      # `Tapioca::Dsl::Compilers::ActiveModelValidationsConfirmation` decorates RBI files for all
+      # classes that use [`ActiveModel::Validates::Confirmation`](https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_confirmation_of).
+      #
+      # For example, with the following class:
+      #
+      # ~~~rb
+      # class User
+      #   include ActiveModel::Validations
+      #
+      #   validates_confirmation_of :password
+      #
+      #   validates :email, confirmation: true
+      # end
+      # ~~~
+      #
+      # this compiler will produce an RBI file with the following content:
+      # ~~~rbi
+      # # typed: true
+      #
+      # class User
+      #
+      #   sig { returns(T.untyped) }
+      #   def email_confirmation; end
+      #
+      #   sig { params(email_confirmation=: T.untyped).returns(T.untyped) }
+      #   def email_confirmation=(email_confirmation); end
+      #
+      #   sig { returns(T.untyped) }
+      #   def password_confirmation; end
+      #
+      #   sig { params(password_confirmation=: T.untyped).returns(T.untyped) }
+      #   def password_confirmation=(password_confirmation); end
+      # end
+      # ~~~
+      class ActiveModelValidationsConfirmation < Compiler
+        extend T::Sig
+
+        ConstantType = type_member do
+          {
+            fixed: T.all(
+              T::Class[ActiveModel::Validations],
+              ActiveModel::Validations::HelperMethods,
+              ActiveModel::Validations::ClassMethods,
+            ),
+          }
+        end
+
+        class << self
+          sig { override.returns(T::Enumerable[Module]) }
+          def gather_constants
+            # Collect all the classes that include ActiveModel::Validations
+            all_classes.select do |c|
+              c < ActiveModel::Validations &&
+                c.name.to_s != "ActiveRecord::Base"
```

Comment:
Why is it necessary to explicitly exclude `ActiveRecord::Base`?

Reply:
Without this the test was returning `ActiveRecord::Base` as a gathered class. I'm assuming it's because ActiveRecord::Base includes ActiveModel::Validations so it gets included there?

Reply:
But does it matter if it is collected? If would just not have any confirmation validations and not do anything, right?

Reply:
That's right. I'll go ahead and remove this then 😄

Reply:
All done! [b78f5bf](https://github.com/Shopify/tapioca/pull/1644/commits/b78f5bf7051bd99c0e73c8e01ec0da58c8258ed4)